File: backend/Models/BudgetPlan.cs
namespace ExpenseTracker.API.Models;

public enum BudgetType { Overall, CreditCard, Food, Transport, Housing, Health, Entertainment, Other }
public enum BudgetPeriod { Weekly, Monthly }

public class BudgetPlan
{
    public int Id { get; set; }
    public BudgetType Type { get; set; }
    public BudgetPeriod Period { get; set; }
    public decimal Limit { get; set; }
}

File: frontend/src/app/models/budget-plan.ts
export type BudgetType = 'Overall' | 'CreditCard' | 'Food' | 'Transport' | 'Housing' | 'Health' | 'Entertainment' | 'Other';
export type BudgetPeriod = 'Weekly' | 'Monthly';

export interface BudgetPlan {
  id: number;
  type: BudgetType;
  period: BudgetPeriod;
  limit: number;
}

export interface BudgetSummary {
  id: number;
  type: BudgetType;
  period: BudgetPeriod;
  limit: number;
  spent: number;
  percentage: number;
}
